django_project/frontend/templates/map.html

```diff
@@ -492,40 +501,51 @@ <h3>activities</h3>
-                    <a class="button edit stroke blue"
-                       style="cursor: pointer"><i
-                            class="fa fa-pencil"></i> edit
-                    </a>
-                </div>
-                <div class="button-wrapper col-xs-4" id="save-button" style="display: none">
-                    <a class="button edit stroke blue"
-                       style="cursor: pointer"><i
-                            class="fa fa-pencil"></i> save
-                    </a>
-                </div>
-                <div class="button-wrapper col-xs-4" id="create-button" style="display: none">
-                    <a class="button edit stroke blue"
-                       style="cursor: pointer"><i
-                            class="fa fa-pencil"></i> create
-                    </a>
-                </div>
-                <div class="button-wrapper col-xs-4" id="add-button">
-                    <a class="button add" style="cursor: pointer">
-                        <i class="fa fa-plus"></i> add new
-                    </a>
-                </div>
-                <div class="button-wrapper col-xs-4" id="cancel-button" style="display: none">
-                    <a class="button edit stroke blue"
-                       style="cursor: pointer"><i
-                            class="fa fa-pencil"></i> cancel
-                    </a>
-                </div>
+                {% if user.social_auth.get.provider == "twitter" %}
+                    <div class="button-wrapper col-xs-4">
+                        <a id="twitter_share_map" class="button share stroke blue">
+                            <i class="fa fa-twitter"></i> share
+                        </a>
+                    </div>
+                {% endif %}
+                {% if user.social_auth.get.provider == "facebook" %}
+                    <div class="fb-share-button"
+                            data-href="https://healthsites.io/map"
+                            data-layout="button_count"
+                            data-size="large">
+                          </div>
+                {% endif %}
+                {% if user.social_auth.get.provider == "openstreetmap" %}
+                    <div class="button-wrapper col-xs-4" id="edit-button">
+                        <a class="button edit stroke blue"
+                           style="cursor: pointer"><i
+                                class="fa fa-pencil"></i> edit
+                        </a>
+                    </div>
+                    <div class="button-wrapper col-xs-4" id="save-button" style="display: none">
+                        <a class="button edit stroke blue"
+                           style="cursor: pointer"><i
+                                class="fa fa-pencil"></i> save
+                        </a>
+                    </div>
+                    <div class="button-wrapper col-xs-4" id="create-button" style="display: none">
```

Comment:
please remove some buttons
just add 3 buttons "edit", "save" and "cancel" (currently it is 4)
we could use javascript to update the text of button